rules/CodeQuality/Rector/Class_/AddReturnDocBlockToCollectionPropertyGetterByToManyAnnotationRector.php

```diff
@@ -0,0 +1,126 @@
+{
+    public function __construct(
+        private readonly ClassMethodReturnTypeOverrideGuard $classMethodReturnTypeOverrideGuard,
+        private readonly PhpDocInfoFactory $phpDocInfoFactory,
+        private readonly DoctrineDocBlockResolver $doctrineDocBlockResolver,
+        private readonly PhpDocTypeChanger $phpDocTypeChanger,
+        private readonly CollectionTypeResolver $collectionTypeResolver,
+        private readonly CollectionTypeFactory $collectionTypeFactory,
+        private readonly MethodUniqueReturnedPropertyResolver $methodUniqueReturnedPropertyResolver,
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Adds @return PHPDoc type to Collection property getter by *ToMany annotation', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+/**
+ * @ORM\Entity
+ */
+final class Trainer
+{
+    /**
+     * @ORM\OneToMany(targetEntity=Training::class, mappedBy="trainer")
+     */
+    private $trainings;
+
+    public function getTrainings()
+    {
+        return $this->trainings;
+    }
+}
+CODE_SAMPLE
+                ,
+                <<<'CODE_SAMPLE'
+/**
+ * @ORM\Entity
+ */
+final class Trainer
+{
+    /**
+     * @ORM\OneToMany(targetEntity=Training::class, mappedBy="trainer")
+     */
+    private $trainings;
+
+    /**
+     * @return \Doctrine\Common\Collections\Collection<int, \Rector\Doctrine\Tests\CodeQuality\Rector\Class_\AddReturnDocBlockToCollectionPropertyGetterByToManyAnnotationRector\Source>
+     */
+    public function getTrainings()
+    {
+        return $this->trainings;
+    }
+}
+CODE_SAMPLE
+            ),
+        ]);
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Class_::class];
+    }
+
+    /**
+     * @param Class_ $node
+     */
+    public function refactorWithScope(Node $node, Scope $scope): ?Node
+    {
+        if (! $this->doctrineDocBlockResolver->isDoctrineEntityClass($node)) {
+            return null;
+        }
+
+        foreach ($node->getMethods() as $classMethod) {
+            if ($this->classMethodReturnTypeOverrideGuard->shouldSkipClassMethod($classMethod, $scope)) {
+                continue;
+            }
+
+            $property = $this->methodUniqueReturnedPropertyResolver->resolve($node, $classMethod);
+
+            if (! $property instanceof Property) {
+                continue;
+            }
+
+            $collectionObjectType = $this->collectionTypeResolver->resolveFromToManyProperties($property);
+
+            if (! $collectionObjectType instanceof FullyQualifiedObjectType) {
+                continue;
+            }
+
+            $phpDocInfo = $this->phpDocInfoFactory->createFromNodeOrEmpty($classMethod);
+            $newVarType = $this->collectionTypeFactory->createType($collectionObjectType);
+            $this->phpDocTypeChanger->changeReturnType($classMethod, $phpDocInfo, $newVarType);
```

Comment:
`$hasChanged` flag is needed here, then, on last, verify `$hasChanged` after loop to return `$node`, otherwise, null returned.

Reply:
thanks, addressed. Added a flag to both AddReturnDocBlockToCollectionPropertyGetterByToManyAnnotationRector and AddReturnDocBlockToCollectionPropertyGetterByToManyAttributeRector